### DISCUSSION TITLE
Using msg.sender outside of a constructor or a function

### DISCUSSION THREAD

**Discussion by urosognjenovic** [Cyfrin/foundry-full-course-cu#1671] [Q&A]
I saw this line of code some time ago and I've been trying to find the answer how this is possible ever since, but to no avail. This contract compiles and the variable owner is indeed set to msg.sender after the contract is deployed.

```solidity
contract A {
    address public owner = msg.sender;
}
```

How is using msg.sender outside of a constructor or a function legitimate? It is mentioned in the Immutability and constants lesson of the course that it is not possible to assign the owner variable to msg.sender outside of a constructor.

**Comment by EngrPips** [comment]
Hello @thebraveman17, Maybe I am wrong, but that piece of code makes sense to me. Any value we set at the top level of a contract body is set at deployment. So whoever send the transaction to deploy the contract would be assigned the `owner`.

**Reply by urosognjenovic** [reply]
Hi, @EngrPips. Thank you for your reply. This line of code will still work even if it is at the bottom of a very long contract. 

Do you know a resource or a documentation that explains this?

**Reply by EngrPips** [reply]
I am convinced the best resource for anything solidity would be the solidity documentation https://docs.soliditylang.org/en/v0.8.25/
